File: RMSofAC.py
import numpy as np

def RMSofAC(inputPPG, index, cycle_head_index, startpoint):
    RMS_AC = np.zeros(10)

    col = np.where(index == cycle_head_index[startpoint])[0][0]
    firstindex = index[col]
    
    for i in range(10):
        data = inputPPG[int(firstindex) + i: int(firstindex) + i + col * 2]  # 1 true cycle
        dataAC = data - np.mean(data)
        RMS_AC[i] = np.sqrt(np.mean(np.square(dataAC)))
    RMS = np.mean(RMS_AC)
    return RMS

File: SpO2/MATLAB/RMSofAC.m
%calculating RMS of AC part and 
function RMS = RMSofAC ( inputPPG, index, cycle_head_index, startpoint )
%inputPPG : raw data after moving average.
%AvRMS : average of 10 RMS values.

RMS_AC = zeros(1, 10);
[row, col] = find( index ==  cycle_head_index(startpoint));
firstindex = index(col);

for i = 1:10
   data = inputPPG(firstindex + i - 1 : firstindex + i); % 1 true cycle
   dataAC = data - mean(data);
   RMS_AC(i) = rms(dataAC);
end

RMS = mean(RMS_AC);

end
